include/net_core_monitor.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef NET_CORE_MONITOR_H_
+#define NET_CORE_MONITOR_H_
+
+/**
+ * @file
+ * @defgroup net_core_monitor Network Core Monitor API
+ * @{
+ * @brief API for the Network Core Monitor.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Checking net_core status.
+ *
+ *  Checking the status of the network core.
+ *  The function should be called less frequently than CONFIG_NCM_APP_FEEDING_INTERVAL_MSEC
+ *
+ *  @param[out]  reset_reas  Reason for network core reset.
```

Comment:
Where to find the relevant reset reasons? Could we refer to the header that contains them?

Reply:
I expanded the description.